project/frontend/src/components/SwipeMatchForm.js

```diff
@@ -22,47 +24,114 @@ function SwipeMatchForm(props) {
   const { currLocation, locationName } = props;
 
   const history = useHistory();
+  const { firestore } = useContext(FirestoreContext);
   const authContext = useContext(AuthContext);
-  const signedIn =
-    authContext.currentUser.get && authContext.currentUser.get.isSignedIn();
+
+  const user = authContext.currentUser.get;
+  const signedIn = user && user.isSignedIn();
 
   const [username, setUsername] = useState('');
   const [groupId, setGroupId] = useState('');
+  const [creatorCurrentSession, setCreatorCurrentSession] = useState(null);
+  const [isGroupIdValid, setIsGroupIdValid] = useState(true);
 
   const anonUsername = useRef('Anonymous ' + Animals.random());
 
   useEffect(() => {
-    const user = authContext.currentUser.get;
-    if (user && user.isSignedIn()) {
+    if (signedIn) {
       setUsername(user.getBasicProfile().getName());
+      (async () => {
+        const creatorCurrentSession = await SwipeMatchService.fetchCreatorCurrentSwipeMatchSession(
+          firestore,
+          user
+        );
+        setCreatorCurrentSession(creatorCurrentSession);
+      })();
     } else {
       setUsername('');
+      setCreatorCurrentSession(null);
     }
-  }, [authContext]);
+  }, [authContext, user, signedIn, firestore]);
 
   const createSession = (event) => {
+    if (event) event.preventDefault();
+
+    (async () => {
+      let id = null;
+
+      try {
+        id = await SwipeMatchService.createSession(firestore, currLocation);
+        SwipeMatchService.updateCreatorCurrentSwipeMatchSession(
+          firestore,
+          user,
+          id
+        );
+      } catch (e) {}
+
+      if (id) {
+        history.push({
+          pathname: '/swipe-match',
+          state: {
+            isCreator: true,
+            username: username ? username : anonUsername.current,
```

Comment:
I believe you could just say `username || anonUsername.current` here because if username is falsy then the value will be the anon username